ext/standard/string.c

```diff
@@ -1883,8 +1883,8 @@ PHP_FUNCTION(strpos)
 		offset += (zend_long)ZSTR_LEN(haystack);
 	}
 	if (offset < 0 || (size_t)offset > ZSTR_LEN(haystack)) {
-		php_error_docref(NULL, E_WARNING, "Offset not contained in string");
-		RETURN_FALSE;
+		zend_value_error("Offset not contained in string");
+		return;
```

Comment:
These are a bit borderline (I wouldn't be surprised if there is code ignoring this warning out there), but overall I think we should do this change.